### PR TITLE
fix: various UI adjustments & fixes (badge details)

### PR DESCRIPTION
**Pull request by daniel-mader**
# Description of change
Solved some UI bugs (mainly on the "Badge details" page).

## Links to any relevant issues
n/a

## How the change has been tested
Manual inspection

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-wallet/src/state/reducers/dev_mode.rs
Comment (Norlock) [inline identity-wallet/src/state/reducers/dev_mode.rs:39]: Maybe we should create a json file to include all these jwt tokens. Maybe outside the scope of this ticket but we can add a TODO comment. So it keeps this file a bit cleaner

File: identity-wallet/src/state/reducers/dev_mode.rs
Comment (Norlock) [inline identity-wallet/src/state/reducers/dev_mode.rs]: Why not use the file function to write (write_all)? https://doc.rust-lang.org/std/fs/struct.File.html

File: unime/src/lib/components/atoms/Image.svelte
Comment (Norlock) [inline unime/src/lib/components/atoms/Image.svelte:74]: Can you not add like this example below: 
```
<h1 class="header {active_class}">Hello reactive variable</h1>
```
If you agree also look at line 62

File: unime/src/routes/(app)/scan/+page@.svelte
Comment (Norlock) [inline unime/src/routes/(app)/scan/+page@.svelte]: Can we maybe put big on click function in the typescript part? To keep the html a bit more clean

File: unime/src/routes/badges/[id]/+page.svelte
Comment (Norlock) [inline unime/src/routes/badges/[id]/+page.svelte]: let issuerId?: string;

File: unime/src/routes/badges/[id]/+page.svelte
Comment (Norlock) [inline unime/src/routes/badges/[id]/+page.svelte:57]: I don't know if you want to modify the existing field, but if you don't maybe use structuredClone(...).

File: unime/src/routes/badges/[id]/+page.svelte
Comment (Norlock) [inline unime/src/routes/badges/[id]/+page.svelte]: Is empty, maybe just short description

File: unime/src/routes/badges/[id]/+page.svelte
Comment (Norlock) [inline unime/src/routes/badges/[id]/+page.svelte]: Maybe create a function with one param entry, so we can give the function a name to convey its meaning.

File: identity-wallet/src/state/persistence.rs
Comment (Norlock) [inline identity-wallet/src/state/persistence.rs:62]: Since this is a lock, don't we need to make this async and use await? Because it can technically fail.

**Review by Norlock — CHANGES_REQUESTED** [review]

File: identity-wallet/src/state/reducers/dev_mode.rs
Comment (Norlock) [inline identity-wallet/src/state/reducers/dev_mode.rs]: This doesn't have to be params but can be read immediately from const.

File: unime/src/routes/badges/[id]/+page.svelte
Comment (SoulaAndrikopoulos) [inline unime/src/routes/badges/[id]/+page.svelte:173]: Did we change "metadata" to "contents" for a specific reason (just curious)?

File: unime/src/routes/badges/[id]/+page.svelte
Comment (SoulaAndrikopoulos) [inline unime/src/routes/badges/[id]/+page.svelte]: Are we providing `alt` text here for the `img`?

File: unime/src-tauri/tests/fixtures/states/three_credentials_redirect_me.json
Comment (nanderstabel) [inline unime/src-tauri/tests/fixtures/states/three_credentials_redirect_me.json:1]: This can now be renamed to `four_credentials_redirect_me.json`

File: identity-wallet/src/utils.rs
Comment (nanderstabel) [inline identity-wallet/src/utils.rs]: Is this still a TODO?

File: unime/src/routes/badges/[id]/+page.svelte
Comment (nanderstabel) [inline unime/src/routes/badges/[id]/+page.svelte]: Feels like something that eventually should be solved in the backend. In any case it could use a comment to add context, e.g:
```typescript
// Custom fields to hide related to NGDIL demo.
const hiddenCustomFields: string[] = ['enrichment'];
```

Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte]: I think every frontend implementation should decide for itself what information it wants to show. If a frontend wants to show the "type" somewhere in the details, why not? If we handled this in the backend, we would take "display logic" to the backend. It would need to be very clear that it is just a "suggestion" to hide these fields, so I'd rather solve that through a good comment instead. I would consider renaming from `hiddenStandardFields` to `omittedStandardFields` or `skippedStandardFields` (but both do not imply visualization whereas "hidden" does) and also add a clear comment of why those fields are omitted when constructing the view. Does that make it clearer?

File: unime/src/routes/badges/[id]/+page.svelte
Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte:173]: Yes, it was decided in a feedback session that "Contents" makes more sense than just the generic "Metadata".

File: unime/src/routes/badges/[id]/+page.svelte
Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte]: I removed it for now since we cannot use it just yet (linking to connections).

File: unime/src/routes/badges/[id]/+page.svelte
Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte:57]: Why? I'm not modifying anything, just unpacking through [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) and assigning it to a new const.

File: unime/src/routes/(app)/scan/+page@.svelte
Comment (daniel-mader) [reply unime/src/routes/(app)/scan/+page@.svelte]: Yes, absolutely. We'll eventually replace all of those external call functions with simple mocks (for UI testing) soon after the new dev profile 🐲 is merged.

File: unime/src/lib/components/atoms/Image.svelte
Comment (daniel-mader) [reply unime/src/lib/components/atoms/Image.svelte:74]: Yes, that's possible but wouldn't work in all cases (CSS cascading). `twMerge()` exists for a reason. [Have a look at this example](https://github.com/dcastil/tailwind-merge/blob/v2.2.1/docs/what-is-it-for.md#thanks-but-i-prefer-to-read) in case you're interested why.

File: unime/src/routes/badges/[id]/+page.svelte
Comment (nanderstabel) [reply unime/src/routes/badges/[id]/+page.svelte]: I think `hiddenCustomFields` is a good name for what it is, but adding a comment would me nice :+1: 

Maybe you're right that indeed what to hide and what not is too subjective to even make suggestions about :+1:

File: identity-wallet/src/utils.rs
Comment (daniel-mader) [reply identity-wallet/src/utils.rs]: Nope, removed it since it's completed. 👍

File: unime/src/routes/badges/[id]/+page.svelte
Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte]: We could make suggestions about it, but probably in a larger comment explaining why we feel those fields are not necessary to be displayed.

File: identity-wallet/src/state/reducers/dev_mode.rs
Comment (nanderstabel) [inline identity-wallet/src/state/reducers/dev_mode.rs]: PERSONAL_INFORMATION **.clone()** .display_credential.id

these don't need to be cloned, same goes for the other credentials :)

File: identity-wallet/src/state/reducers/dev_mode.rs
Comment (daniel-mader) [reply identity-wallet/src/state/reducers/dev_mode.rs]: I extracted a helper function and switched to `file.write_all()` 👍

File: unime/src/routes/(app)/scan/+page@.svelte
Comment (Norlock) [inline unime/src/routes/(app)/scan/+page@.svelte]: this url will change so remove, maybe you can wait for my PR I will add steps for dev profile

Comment (daniel-mader) [reply unime/src/routes/(app)/scan/+page@.svelte]: Actually, I would like to get this PR here merged as soon as possible since it contains important UI bug fixes. The PR #117 you mentioned is a great improvement for developer experience, but it's not crucial for most clients who just want to try out the app.

**Review by nanderstabel — DISMISSED** [review]

File: unime/src/routes/(app)/scan/+page@.svelte
Comment (Norlock) [reply unime/src/routes/(app)/scan/+page@.svelte]: Yes but it doesn't fix that the url is not static ;), so maybe you can wait for the other branch to be merged first and then call that function or a similar one.

Comment (daniel-mader) [reply unime/src/routes/(app)/scan/+page@.svelte]: I agree and I removed the static URL completely 👍

File: unime/src/routes/badges/[id]/+page.svelte
Comment (Norlock) [inline unime/src/routes/badges/[id]/+page.svelte:183]: Last thing then I will approve this PR, maybe call it printBadgeDetails(), or something like that so we now what it prints

Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte:183]: I think a function name should reflect what it does. The name is purposely generic since it is agnostic about its input, so I would not like to make the name too specific. The comment should add the necessary context, don't you think?

```ts
// TODO: this is a simple way to display any (potentially nested) data, since we don't have a proper UI design for it yet
const prettyPrint = (object: any): string => {
  return JSON.stringify(object, null, 2);
};
```

File: unime/src/routes/badges/[id]/+page.svelte
Comment (daniel-mader) [reply unime/src/routes/badges/[id]/+page.svelte]: Duplicate of https://github.com/impierce/identity-wallet/pull/115#discussion_r1492391882

**Review by Norlock — APPROVED** [review]
LGTM

**Review by nanderstabel — APPROVED** [review]

**Pull request merged**
